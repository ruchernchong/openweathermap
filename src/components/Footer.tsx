import React from "react";
import styled from "styled-components";
import { colours } from "../theme";
import { graphql, useStaticQuery } from "gatsby";

const StyledFooter = styled.footer`
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 32px 0;
  box-shadow: 0 -5px 5px -5px ${colours.gray80};
`;

export const Footer = () => {
  const data = useStaticQuery(graphql`
    {
      site {
        siteMetadata {
          author
          version
        }
      }
    }
  `);

  const { author, version } = data.site.siteMetadata;

  return (
    <StyledFooter>
      {`Â©${new Date().getFullYear()} Developed by ${author}, ${version}`}
    </StyledFooter>
  );
};
