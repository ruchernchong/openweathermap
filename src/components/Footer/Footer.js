import React from "react";
import styled from "styled-components";
import { colours } from "../../theme";

import { author, version } from "../../../package.json";

const StyledFooter = styled.footer`
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 32px 0;
  box-shadow: 0 -5px 5px -5px ${colours.gray80};
`;

const Footer = () => {
  return (
    <StyledFooter>
      {`Â©${new Date().getFullYear()} Developed by ${author}, v${version}`}
    </StyledFooter>
  );
};

export default Footer;
