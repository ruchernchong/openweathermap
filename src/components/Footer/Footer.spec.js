import React from "react";
import { render } from "@testing-library/react";
import Footer from "./Footer";

describe("Footer", () => {
  test("should render", () => {
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });
});
