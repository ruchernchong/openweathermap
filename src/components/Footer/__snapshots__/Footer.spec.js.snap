// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render 1`] = `
<div>
  <footer
    class="sc-AxjAm gWcbgK"
  >
    Â©2020 Developed by Chong Ru Chern, v0.1.0
  </footer>
</div>
`;
